<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>The Extended Integer (XInt) Library: The \c on_exception Function And \c -fno-exceptions Support</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
  <div class="navpath">
    <ul>
      <li><a class="el" href="index.html">The Extended Integer (XInt) Library</a>      </li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>The <code>on_exception</code> Function And <code>-fno-exceptions</code> Support </h1>  </div>
</div>
<div class="contents">
<p>This library includes support for GCC's <code>-fno-exceptions</code> option. When a program is compiled with <code>-fno-exceptions</code>, it cannot use <code>try</code>, <code>throw</code>, or <code>catch</code> statements, and any error will result in a call to <code>abort()</code> unless otherwise dealt with. To activate this support, define the preprocessor macro <code>BOOST_XINT_NO_EXCEPTIONS</code> before including any Boost.XInt header.</p>
<p>To give your program a chance to at least record any error, and optionally recover from it using the <code>setjmp/longjmp</code> functions, the library includes an <a class="el" href="namespaceboost_1_1xint.html#ac2b936b22084368c66e62f8181c08241">on_exception</a> function. You call this function, passing in a function that you want called whenever there is a problem. The library will call that function and (if it returns) throw the exception as normal. (If BOOST_XINT_NO_EXCEPTIONS is used, it will call <code>abort()</code> after calling your function instead.)</p>
<p>Note that while <code>setjmp/longjmp</code> may allow you to recover from errors, it will often result in memory leaks since no destructors or other cleanup code is allowed to run. </p>
</div>
<hr>
<p><i>&copy; Copyright Chad Nelson, 2010-2011. Distributed under the Boost Software
License, Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
<a href=http://www.boost.org/LICENSE_1_0.txt>http://www.boost.org/LICENSE_1_0.txt</a>)
</i></p>
